pkg/operator/controller/dns/controller_test.go

```diff
@@ -0,0 +1,93 @@
+package dns
+
+import (
+	"fmt"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+	"testing"
+)
+
+func TestPublishRecordToZones(t *testing.T) {
+	var tests = []struct {
+		name   string
+		record *iov1.DNSRecord
+		zones  []configv1.DNSZone
+		expect operatorv1.ConditionStatus
+	}{
+		{
+			name: "testing for a successful update of 1 record",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zones: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			expect: operatorv1.ConditionFalse,
+		},
+		{
+			name: "when no zones available",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zones:  []configv1.DNSZone{{}},
+			expect: operatorv1.ConditionTrue,
+		},
+		{
+			name:   "when no region available",
+			record: nil,
+			zones: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			expect: operatorv1.ConditionTrue,
+		},
+	}
+	for _, test := range tests {
+		r := &reconciler{
+			dnsProvider: &FakeDNSProvider{},
+		}
+		t.Logf("evaluating test %s", test.name)
+		actual, results := r.publishRecordToZones(test.zones, test.record)
+		_ = results
+		for _, dnsStatus := range actual {
+			for _, condition := range dnsStatus.Conditions {
+				if condition.Status != string(test.expect) {
+					t.Fatalf("expected:\n%#v\ngot:\n%#v", test.expect, condition.Status)
+				}
+			}
+		}
+	}
+}
+
+var _ dns.Provider = &FakeDNSProvider{}
+
+type FakeDNSProvider struct{}
```

Comment:
I meant to use the existing `FakeProvider`.  Will that suffice?

Reply:
But then I will need condition  in the existing one right [0]   if zone.ID == "" {
		return fmt.Errorf("zone is required")} else {
		return nil
	}  
?
[0] https://github.com/openshift/cluster-ingress-operator/blob/a140e2d5347a4503d263eadf9e0724a87eb73dc6/pkg/dns/dns.go#L22

Reply:
I think this will not be needed once https://github.com/openshift/cluster-ingress-operator/pull/328/files#r351373838 is addressed.

Reply:
Oh, we need this for the "when one zone available and one not available" test case.  Never mind!

Reply:
I am now using FakeProvider